src/logic.js

```diff
@@ -1,14 +1,22 @@
 define([
-    'vellum/util',
     'xpath',
     'xpathmodels',
     'underscore'
 ], function (
-    util,
     xpath,
     xpathmodels,
     _
 ) {
+    var XPATH_REFERENCES = [
+        "relevantAttr",
+        "calculateAttr",
+        "constraintAttr",
+        "dataParent",
+        "repeat_count",
+        "filter",
+        "defaultValue"
+    ], NO_SELF_REFERENCES = _.without(XPATH_REFERENCES, 'constraintAttr');
```

Comment:
Just noticing that previously self references only produced a warning in [labels and `calculateAttr` and `relevantAttr`](https://github.com/dimagi/Vellum/commit/4c670bb53b05238fa3374538e24ba6d3b53505f1#diff-7eb52b366866677666470e019283c8eaL1592). So this is new to be warning for the others here. I suppose they all make sense though.